locator/token_metadata.cc

```diff
@@ -1547,7 +1547,7 @@ void token_metadata_impl::calculate_pending_ranges_for_leaving(
     auto affected_ranges_size = affected_ranges.size();
     tlogger.debug("In calculate_pending_ranges: affected_ranges.size={} stars", affected_ranges_size);
     for (const auto& r : affected_ranges) {
-        auto t = r.end() ? r.end()->value() : dht::maximum_token();
+        auto t = r.end() ? r.end()->value() : dht::smallest_token();
```

Comment:
s/smallest/greatest/

Reply:
No, this is not a mistake. `calculate_natural_endpoints` starts the search (in `token_metadata_impl::_sorted_tokens`) from the passed token, *inclusive*. So passing `maximum_token` actually wraps around and starts from the smallest token. Passing `greatest_token` would start from INT64_MAX, which would change the behaviour if `INT64_MAX` is in `sorted_tokens`.

Reply:
Ah. The wrap-around behavior is so confusing and terrible.